converters/schema.ts

```diff
@@ -0,0 +1,75 @@
+import { ClarityDataSchema, ObjectType } from "../clarity";
+import { hashCode } from "./../src/utils";
+
+export class SchemaManager {
+
+  private schemas = {};
+
+  public getSchema(schemaHash: string): ClarityDataSchema {
+    return this.schemas[schemaHash];
+  }
+
+  public addSchema(schema: ClarityDataSchema): boolean {
+    let schemaHash = hashCode(JSON.stringify(schema));
```

Comment:
>hashCode [](start = 21, length = 8)

What if there is a collision?

Reply:
I switched to hand-managed schema ids - will be shorted than hash code over the wire and no collisions possible.